### PR TITLE
Replace Quickcheck with Autocheck.

### PR DESCRIPTION
**Pull request by endJunction**
Testing compilation on other plattforms.

In the first step the quickcheck used in the tests of BaseLib.Quicksort is replaced by autocheck.
In the second step I exemplary applied autocheck to tests of `Point3d` operators `operator==()` `operator<()` and `lessEq()`

### CODE REVIEW

**Comment by endJunction** [comment]
Jenkins, test this please!

**Comment by endJunction** [comment]
@bilke Jenkins still tries to checkout the 'autocheck' via git interface, not https... ;(

**Comment by bilke** [comment]
Linux now fails in the benchmark runs. Windows failed because of a job configuration error.

Jenkins test this please!

**Comment by endJunction** [comment]
Jenkins, test this!

**Comment by endJunction** [comment]
Jenkins test this please!

**Comment by endJunction** [comment]
Jenkins test this please!

**Comment by bilke** [comment]
Cool stuff! :+1:

File: Tests/MathLib/TestPoint3d.cpp [hunk 1]
```diff
@@ -11,184 +11,220 @@
...
+		q[coordinate] = q[coordinate] + perturbation;
+		return lessEq(p, q) && lessEq(q, p);
+	};
+
+	auto eps = std::numeric_limits<double>::epsilon();
+
+	ac::check<MathLib::Point3d, double, unsigned>(
+		pointWithSmallAddedValue, 10000,
+		ac::make_arbitrary(pointGenerator,
+						   ac::progressivelySmallerGenerator<double>(eps / 2),
```
Comment (TomFischer) [inline Tests/MathLib/TestPoint3d.cpp]: The spacing looks wired to me.

File: Tests/MathLib/TestPoint3d.cpp [hunk 2]
```diff
@@ -11,184 +11,220 @@
...
-	ASSERT_TRUE((lessEq(a,b) && lessEq(b,a)) == (a == b));
+	// A point is always equal to itself and its copy.
+	auto samePointEqualCompare = [](MathLib::Point3d const& p)
+	{
+		auto q = p;
+		return (p == p) && (p == q) && (q == p);
+	};
+
+	ac::check<MathLib::Point3d>(samePointEqualCompare, 100,
+								ac::make_arbitrary(pointGenerator),
```
Comment (TomFischer) [inline Tests/MathLib/TestPoint3d.cpp]: The spacing looks wired to me.

File: Tests/MathLib/TestPoint3d.cpp [hunk 3]
```diff
@@ -11,184 +11,220 @@
...
+	ac::check<MathLib::Point3d, double, unsigned>(
+		pointWithLargeAddedValue, 10000,
+		ac::make_arbitrary(pointGenerator, ac::generator<double>(),
+						   coordGenerator)
+			.discard_if(
+				[&eps](MathLib::Point3d const&, double const v, unsigned const)
+				{
+					return !(v > eps);
+				}),
+		gtest_reporter);
```
Comment (TomFischer) [inline Tests/MathLib/TestPoint3d.cpp]: The spacing looks wired to me.

File: Tests/MathLib/TestPoint3d.cpp [hunk 4]
```diff
@@ -11,184 +11,220 @@
...
+	};
+
+	auto eps = std::numeric_limits<double>::epsilon();
+
+	ac::check<MathLib::Point3d, double, unsigned>(
+		pointWithSmallAddedValue, 1000,
+		ac::make_arbitrary(pointGenerator,
+						   ac::progressivelySmallerGenerator<double>(eps / 2),
+						   coordGenerator),
+		gtest_reporter);
```
Comment (TomFischer) [inline Tests/MathLib/TestPoint3d.cpp]: The spacing looks wired to me.

File: Tests/MathLib/TestPoint3d.cpp [hunk 5]
```diff
@@ -11,184 +11,220 @@
...
+	// A point is never less than itself or its copy.
+	auto samePointLessCompare = [](MathLib::Point3d const& p)
+	{
+		auto q = p;
+		return !(p < p) && !(p < q) && !(q < p);
+	};
+
+	ac::check<MathLib::Point3d>(samePointLessCompare, 100,
+								ac::make_arbitrary(pointGenerator),
+								gtest_reporter);
```
Comment (TomFischer) [inline Tests/MathLib/TestPoint3d.cpp]: The spacing looks wired to me.

File: Tests/MathLib/TestPoint3d.cpp [hunk 6]
```diff
@@ -11,184 +11,220 @@
...
+	ac::check<MathLib::Point3d, double, unsigned>(
+		pointWithAddedValue, 1000,
+		ac::make_arbitrary(pointGenerator,
+						   ac::map(&ac::absoluteValue, ac::generator<double>()),
+						   coordGenerator)
+			.discard_if(
+				[](MathLib::Point3d const&, double const eps, unsigned const)
+				{
+					return eps == 0;
+				}),
```
Comment (TomFischer) [inline Tests/MathLib/TestPoint3d.cpp]: The spacing looks wired to me.

File: Tests/MathLib/TestPoint3d.cpp [hunk 7]
```diff
@@ -11,184 +11,220 @@
...
+		pointWithSubtractedValue, 1000,
+		ac::make_arbitrary(pointGenerator,
+						   ac::map(&ac::absoluteValue, ac::generator<double>()),
+						   coordGenerator)
+			.discard_if(
+				[](MathLib::Point3d const&, double const eps, unsigned const)
+				{
+					return eps == 0;
+				}),
+		gtest_reporter);
```
Comment (TomFischer) [inline Tests/MathLib/TestPoint3d.cpp]: The spacing looks wired to me.

**Comment by TomFischer** [comment]
Apart from the formating issues: :+1:

File: Tests/MathLib/TestPoint3d.cpp (hunk 1 above)
Comment (endJunction) [inline Tests/MathLib/TestPoint3d.cpp]: Looks good with tabwidth=4. I'd prefer spaces (of course) but the original file had tabs in it...
formatting

Comment (bilke) [inline Tests/MathLib/TestPoint3d.cpp]: indent with **tabs**, align with **spaces**...

**Comment by endJunction** [comment]
Reformatted and rebased.

**Pull request merged**